### PR TITLE
Remove book notebook from toc

### LINKED ISSUE #14116: Close jupyter notebook command does not work in notebooks within a book.
The close jupyter notebook command does not work in notebooks within a book.
- Azure Data Studio Version:
Version: 1.25.2
Commit: 4d4917d328ab7a7526817a0f0ef03237a79aa54d
Date: 2021-01-20T22:47:41.882Z
VS Code: 1.48.0
Electron: 9.3.0
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 20.2.0
Steps to Reproduce:

1. Open a book, right click "Close Jupyter Notebook".
2. It doesn't do anything.

Feature request: 
Replace command for remove notebook which removes the notebook from the toc.yml.

### PR DESCRIPTION
**Pull request by barbaravaldez**
This PR fixes #14116 
Book notebooks:
Screen Shot 2021-03-12 at 11 00 28 AM

Standalone notebooks:
Screen Shot 2021-03-12 at 11 00 50 AM

### CODE REVIEW

File: extensions/notebook/package.json [hunk 1]
```diff
@@ -465,13 +473,17 @@
...
           "when": "view == bookTreeView && viewItem == savedNotebook"
         },
+        {
+          "command": "notebook.command.removeNotebook",
+          "when": "view == bookTreeView && viewItem == savedBookNotebook"
+        },
         {
           "command": "notebook.command.moveTo",
-          "when": "view == bookTreeView && viewItem == savedNotebook || view == bookTreeView && viewItem == section"
+          "when": "view == bookTreeView && viewItem == savedNotebook || view == bookTreeView && viewItem == savedBookNotebook || view == bookTreeView && viewItem == section"
```
Comment (lucyzhang929) [inline extensions/notebook/package.json:482]: Can this be shorted to `"view == bookTreeView && (viewItem == savedNotebook || viewItem == savedBookNotebook || viewItem == section)"`

File: extensions/notebook/package.json [hunk 2]
```diff
@@ -465,13 +473,17 @@
...
         },
         {
           "command": "notebook.command.pinNotebook",
-          "when": "view == bookTreeView && viewItem  == savedNotebook",
+          "when": "view == bookTreeView && viewItem  == savedNotebook || view == bookTreeView && viewItem == savedBookNotebook",
```
Comment (lucyzhang929) [inline extensions/notebook/package.json:486]: Similar to above `"view == bookTreeView && (viewItem == savedNotebook || viewItem == savedBookNotebook)"`

**Review by lucyzhang929 — APPROVED** [review]

File: extensions/notebook/src/book/bookTocManager.ts [hunk 3]
```diff
@@ -426,11 +424,16 @@ export class BookTocManager implements IBookTocManager {
...
 				await fs.writeFile(targetBook.tableOfContentsPath, yaml.safeDump(this.tableofContents, { lineWidth: Infinity, noRefs: true, skipInvalid: true }));
 			} else {
-				await this.updateTOC(targetBookVersion, targetBook.tableOfContentsPath, targetSection, this.newSection);
+				await this.updateTOC(targetBook.book.version, targetBook.tableOfContentsPath, targetSection, this.newSection);
 			}
 		}
 	}
 
+	public async removeNotebook(element: BookTreeItem): Promise<void> {
+		const findSection = { file: element.book.page?.file?.replace(/\\/g, '/'), title: element.book.page?.title };
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTocManager.ts:433]: I seem to see a lot of this replacing of backslashes in the path - is this not something that would be better done once when you load the book and then reuse everywhere that needs it?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 4]
```diff
@@ -276,6 +276,10 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		}
 	}
 
+	async removeNotebook(bookItem: BookTreeItem): Promise<void> {
+		await this.bookTocManager.removeNotebook(bookItem);
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:280]: You can just return the call directly - no need to await here (that's just unnecessary overhead)

File: extensions/notebook/src/common/utils.ts [hunk 5]
```diff
@@ -441,6 +442,15 @@ export function isBookItemPinned(notebookPath: string): boolean {
 	return false;
 }
 
+export function getNotebookType(book: BookTreeItemFormat): string {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/common/utils.ts:445]: Should the return type be BookTreeItemType?

File: extensions/notebook/src/extension.ts [hunk 6]
```diff
@@ -47,6 +47,7 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
 	extensionContext.subscriptions.push(vscode.commands.registerCommand('notebook.command.openBook', () => bookTreeViewProvider.openNewBook()));
 	extensionContext.subscriptions.push(vscode.commands.registerCommand('notebook.command.closeBook', (book: any) => bookTreeViewProvider.closeBook(book)));
 	extensionContext.subscriptions.push(vscode.commands.registerCommand('notebook.command.closeNotebook', (book: any) => bookTreeViewProvider.closeBook(book)));
+	extensionContext.subscriptions.push(vscode.commands.registerCommand('notebook.command.removeNotebook', (book: any) => bookTreeViewProvider.removeNotebook(book)));
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/extension.ts:50]: Generally it's good to give the params the types that you expect. So BookTreeItem in this case. 

(granted anyone can call this at any time and pass in whatever, but that isn't something I generally see value in defending against. But at the very least informing people looking at the code as to what you expect it to be is useful)

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/notebook/package.json (hunk 2 above)
Comment (barbaravaldez) [reply extensions/notebook/package.json:486]: I tried that but the `when` clause doesn't seem to support parentheses for grouping conditions 😞. If I use it then it only takes into account the last OR.
https://code.visualstudio.com/api/references/when-clause-contexts#conditional-operators

**Pull request merged**